### PR TITLE
[search] Correct check for RankTable validity.

### PR DESCRIPTION
**Pull request by deathbaba**
Fixes assert in Debug with old MWM files.

### CODE REVIEW

**Comment by vng** [comment]
LGTM

File: indexer/rank_table.cpp [hunk 1]
```diff
@@ -221,7 +221,11 @@ void SerializeRankTable(RankTable & table, string const & mapPath)
 template <typename TRegion>
 unique_ptr<RankTable> LoadRankTable(unique_ptr<TRegion> && region)
 {
-  if (!region || !region->ImmutableData() || region->Size() < kHeaderSize)
+  if (!region || !region->ImmutableData())
+  {
+    return unique_ptr<RankTable>();
+  }
+  else if (region->Size() < kHeaderSize)
```
Comment (ygorshenin) [inline indexer/rank_table.cpp]: no else after return, please.

**Comment by ygorshenin** [comment]
What's the point in this change?

**Comment by ygorshenin** [comment]
I mean, could y

**Comment by ygorshenin** [comment]
ou please describe a case where old check fails?

File: indexer/rank_table.cpp (hunk 1 above)
Comment (deathbaba) [inline indexer/rank_table.cpp]: ?

**Comment by deathbaba** [comment]
We just discussed it in Skype chat. Old code fails with old MWMs in debug when you try to search for anything.

**Comment by deathbaba** [comment]
@ygorshenin PTAL please.

File: indexer/rank_table.cpp (hunk 1 above)
Comment (ygorshenin) [inline indexer/rank_table.cpp]: Please, rewrite the code like:
```c++
if (...)
  return make_unique<RankTable>();
if (...)
```

Comment (deathbaba) [inline indexer/rank_table.cpp]: Already done!

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**